app/main.py

```diff
@@ -1,3 +1,63 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, run: int | float) -> None:
+        self.km = run
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            total_distance = self.km + other.km
+        else:
+            total_distance = self.km + other
+        result = Distance(total_distance)
+        return result
```

Comment:
There is no need for the `result` variable, you should return a `Distance` class instance directly here and in other places to not allocate additional memory to store this instance.